chainer/functions/array/where.py

```diff
@@ -26,11 +26,13 @@ def forward(self, inputs):
         condition, x, y = inputs
         return xp.where(condition, x, y),
 
-    def backward(self, inputs, grads):
-        xp = cuda.get_array_module(inputs[0])
-        condition = inputs[0]
-        gx = xp.where(condition, grads[0], 0)
-        gy = xp.where(condition, 0, grads[0])
+    def backward(self, indexes, grad_outputs):
+        condition = self.inputs[0]
+        xp = cuda.get_array_module(condition.data)
```

Comment:
You have to call `self.retain_inputs` if you need input data, and use `get_retained_inputs` to obtain the input variable (see the implementation of `linear` for example).

Reply:
Fixed.
Some other functions are directly using self.input and they must be fixed too I think?
e.g., https://github.com/kmaehashi/chainer/blob/492c9bb61fd44457c5e81df5c87120807110e7b8/chainer/functions/array/permutate.py#L79